pdp/static/js/pdp_controls.js

```diff
@@ -204,15 +234,16 @@ Colorbar.prototype = {
     },
 
     redraw: function () {
+        this.midpoint = this.calculate_midpoint();
+        this.apply_precision(this.estimate_precision());
         var div = $("#" + this.div_id);
         div.css('background-image', "url(" + this.graphic_url() + ")");
-        div.find("#minimum").html(round(this.minimum) + " " + this.units);
-        div.find("#maximum").html(round(this.maximum) + " " + this.units);
-        div.find("#midpoint").html(round(this.midpoint()) + " " + this.units);
+        div.find("#minimum").html(this.minimum + " " + this.units);
```

Comment:
In these three lines, I'd rather you apply the precision locally at render-time than modify it in place and then render the modified version.